function-controller/Makefile

```diff
@@ -20,11 +20,18 @@ gen-mocks:
 clean:
 	rm -f $(OUTPUT)
 
-dockerize: $(GO_SOURCES)
+dockerize: $(GO_SOURCES) ../vendor
 	docker build .. --build-arg COMPONENT=function-controller -t projectriff/function-controller:$(TAG)
 
-debug-dockerize: $(GO_SOURCES)
+debug-dockerize: $(GO_SOURCES) ../vendor
 	# Need to remove probes as delve starts app in paused state
 	-kubectl patch deploy/function-controller --type=json -p='[{"op":"remove", "path":"/spec/template/spec/containers/0/livenessProbe"}]'
 	-kubectl patch deploy/function-controller --type=json -p='[{"op":"remove", "path":"/spec/template/spec/containers/0/readinessProbe"}]'
 	docker build . -t projectriff/function-controller:$(TAG) -f Dockerfile-debug
+
+../vendor: ../glide.lock
+	$(MAKE) -C .. vendor
+
+../glide.lock: ../glide.yaml
```

Comment:
This allows a “top level” make operation to be initiated from the component makefile. This could result in undefined behaviour if component makefiles are invoked concurrently, but I guess that's unlikely to happen in practice.

Reply:
yeah, maybe I overused recursive `make` here.
Maybe just `cd .. && glide up -v etc...`?